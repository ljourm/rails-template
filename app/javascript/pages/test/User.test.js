import { createLocalVue, mount } from "@vue/test-utils"
import Vuex from "vuex"
import Component from "../User"

const email = "test@example.com"
const user = { uuid: "uuid", email: email, name: "name", roles: [] }

describe("Component", () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)

  const state = { user: user }
  const store = new Vuex.Store({
    modules: {
      user: {
        namespaced: true,
        state,
      },
    },
  })

  const wrapper = mount(Component, {
    localVue,
    store,
  })

  it("snapshot", () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
