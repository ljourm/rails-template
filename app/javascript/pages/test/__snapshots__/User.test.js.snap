// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot 1`] = `
<form>
  <div
    class="field"
  >
    <label
      class="label"
    >
      氏名
    </label>
    <div
      class="control"
    >
      <input
        class="input"
        type="text"
      />
    </div>
  </div>
  <div
    class="field"
  >
    <label
      class="label"
    >
      EMail
    </label>
    <div
      class="control"
    >
      <input
        class="input"
        type="text"
      />
    </div>
  </div>
  <div
    class="field"
  >
    <label
      class="label"
    >
      パスワード
    </label>
    <div
      class="control"
    >
      <input
        class="input"
        type="password"
      />
    </div>
  </div>
  <div
    class="field"
  >
    <label
      class="label"
    >
      パスワード確認
    </label>
    <div
      class="control"
    >
      <input
        class="input"
        type="password"
      />
    </div>
  </div>
  <div
    class="field is-grouped"
  >
    <div
      class="control"
    >
      <button
        class="button destroy"
      >
        削除
      </button>
    </div>
    <div
      class="control"
    >
      <button
        class="button is-primary submit"
        type="submit"
      >
        送信
      </button>
    </div>
  </div>
</form>
`;
